framework/src/kernels/Kernel.C

```diff
@@ -80,6 +80,25 @@ Kernel::computeJacobian()
 
   ke += _local_ke;
 
+  if (_calculate_nonlocal_variable_jacobian)
+  {
+    _nonlocal_ke.resize(ke.m(), _subproblem.es().n_dofs());
+    _nonlocal_ke.zero();
+    for (_i = 0; _i < _test.size(); _i++)
```

Comment:
Well, this is not working as expected yet. The jacobianblock in assembly needs to be sized with all dofs corresponding to each variable, instead of local dofs, that might make it even slower. I am open to ideas about smarter ways of doing this.